packages/flutter/lib/src/widgets/undo_history.dart

```diff
@@ -178,9 +187,9 @@ class UndoHistoryState<T> extends State<UndoHistory<T>> with UndoManagerClient {
       return;
     }
 
-    _lastValue = widget.value.value;
+    _lastValue = widget.undoStackModifier?.call(widget.value.value) ?? widget.value.value;
 
-    _throttleTimer = _throttledPush(widget.value.value);
+    _throttleTimer = _throttledPush(_lastValue as T);
```

Comment:
Does this dedup like the ValueNotifier class does? If you map all inputs to the same value, does it push one entry to the stack, or multiple entries that are all identical?

Reply:
In that case it would just push one entry.

https://github.com/flutter/flutter/blob/db7ef5bf9f59442b0e200a90587e8fa5e0c6336a/packages/flutter/lib/src/widgets/undo_history.dart#L169-L171

Reply:
Oh I was mistaken, this happens before the modifier gets a chance to do anything. I can make sure the modified value is taken into account for this logic.